### PR TITLE
chore(topology): Update shutdown deadlines

### LINKED ISSUE #2109: Update all timeouts related to shutting down sources
[Issue 1091](https://github.com/timberio/vector/issues/1091) created a SourceShutdownManager which is responsible for sending the shutdown signal to all sources, and then triggering force shutdown if they fail to shut down gracefully within a given deadline.  Since currently most sources are not yet looking for the new shutdown signal, the timeouts before issuing force shutdown have been kept quite low.  Once [issue 2108](https://github.com/timberio/vector/issues/2108) is resolved, however, we should increase all these timeouts as we expect sources to promptly shut down gracefully.

Places where these timeouts need to be changed can be found by looking for the callers of the SourceShutdownManager's `shutdown_all` and `shutdown_source` methods in topology/mod.rs.

### PR DESCRIPTION
**Pull request by ktff**
Closes  #2109.

Blocked on ~~#2108~~ #2551.

### CODE REVIEW

File: src/topology/mod.rs [hunk 1]
```diff
@@ -288,16 +282,14 @@ impl RunningTopology {
 
         // First pass to tell the sources to shut down.
         let mut source_shutdown_complete_futures = Vec::new();
-        // TODO: Once all Sources properly look for the ShutdownSignal, up this time limit to something
-        // more like 30-60 seconds.
 
         // Only log that we are waiting for shutdown if we are actually removing
         // sources.
         if !diff.sources.to_remove.is_empty() {
             info!("Waiting for up to 3 seconds for sources to finish shutting down");
```
Comment (binarylogic) [inline src/topology/mod.rs:289]: Do we need to change this message? I also think 30 seconds is plenty here.

Comment (ktff) [reply src/topology/mod.rs]: Indeed.

File: src/topology/mod.rs [hunk 2]
```diff
@@ -179,14 +179,8 @@ impl RunningTopology {
             .map(|_| ())
             .map_err(|_| ());
 
-        // TODO: Once all Sources properly look for the ShutdownSignal, remove this in favor of
-        // using 'deadline' instead.
```
Comment (lukesteensen) [inline src/topology/mod.rs:183]: Is there a way of verifying that this is the case?

Comment (ktff) [reply src/topology/mod.rs:183]: If there is a source not looking for the `ShutdownSignal`, it should prolong the shutdown for that much. So a test to measure the shutdown length of vector with each/every source and seeing if it's less than the deadline, would do the trick. I'll extend #1089 with it.    

---
Good thing you asked, our stop procedure is still dropping the handles of source tasks, so we aren't actually using `ShutdownSignal` yet. I'll add an issue for it.

**Pull request merged**